app/controllers/topics_controller.rb

```diff
@@ -8,6 +8,38 @@ def index
     end
   end
 
+  def vote
+    @topic = Topic.find params[:id]
+    @topic.votes += 1
```

Comment:
This is a lot better than doing it in the browser because the distance to the actual operation is shorter and reduces the risk of a race condition, but it's still there. To make it completely atomic, we should use PostgreSQL to deduct from the count. In SQL, that would be:

`UPDATE topics SET votes=votes-1`

I actually haven't done that in ActiveRecord before, though, so I'm not sure what the syntax would be.